tests/phpunit/CRM/Contribute/BAO/ContributionRecurTest.php

```diff
@@ -564,4 +565,32 @@ public function yearMonthProvider() {
     ];
   }
 
+  /**
+   * Test Recurring Contribution Email Receipt Flag
+   *
+   * @throws \CRM_Core_Exception
+   */
+  public function testContributionEmailReceipt(): void {
+    $createParams = $this->_params;
+    unset($createParams['trxn_id'], $createParams['invoice_id']);
+
+    // pass null value to is_email_receipt
+    $createParams['is_email_receipt'] = NULL;
+    $recurring1 = CRM_Contribute_BAO_ContributionRecur::add($createParams);
```

Comment:
I think the test failure is just that the add() function doesn't take this type of params. The params format seems intended for the api which converts relative dates like 'yesterday' into actual dates. If you give that to add() then it tries to literally insert 'yesterday' into a date field.

Reply:
Ok - I fixed it up to call the api - hopefully it passes now